src/components/App/App.tsx

```diff
@@ -58,18 +58,22 @@ class App extends React.Component<Props> {
   };
 
   toggleFavorite = (ApiCard: Api) => {
-    const alreadySaved = this.state.favorites.some((element) => element.API === ApiCard.API);
-    if(!alreadySaved){
+    const isSaved = this.state.favorites.some((element) => element.API === ApiCard.API);
+    if(!isSaved){
       this.setState({ favorites: [...this.state.favorites, ApiCard] });
     } else {
       const updatedFavorites = this.state.favorites.filter(element => element.API !== ApiCard.API)
       this.setState({ favorites: updatedFavorites})
     }
+    setTimeout(() => {
+      !this.state.favorites.length && localStorage.setItem('favorites', JSON.stringify(this.state.favorites));
+    },0)
   };
 
```

Comment:
What does this setTimeout function do? Just interested

Reply:
It forces this statement into a queue so that the state has time to finish updating before this runs.